pkg/image/resolve.go

```diff
@@ -61,20 +59,72 @@ func getChartAndVersion(path string) (map[string]chart, error) {
 	if err != nil {
 		return nil, err
 	}
-	for k, versions := range index.IndexFile.Entries {
-		// because versions is sorted in reverse order, the first one will be the latest version
+	for _, versions := range index.IndexFile.Entries {
 		if len(versions) > 0 {
-			newestVersionedChart := versions[0]
-			rtn[k] = chart{
-				dir:     newestVersionedChart.Dir,
-				version: newestVersionedChart.Version}
+			// Add latest version (First element is latest, list sorted in descending order)
+			chartVersions = append(chartVersions, versions[0])
+			if strings.Contains(path, SystemChartsRepoDir) {
+				// Skip if rancherVersion is not a valid semver (E.g is a commit hash)
+				if rancherSemVer, err := semver.NewVersion(rancherVersion); err == nil {
+					// Skip first element, the latest version has already been added
+					for _, v := range versions[1:] {
+						questions, err := getQuestions(filepath.Join(path, v.Dir))
+						if err != nil {
+							return nil, fmt.Errorf("no questions file in chart %s:%s", v.Name, v.Version)
+						}
+						min, ok := questions["rancher_min_version"].(string)
+						if !ok {
+							return nil, fmt.Errorf("no rancher_min_version set in chart %s:%s", v.Name, v.Version)
+						}
+						// No ok check, max is optional
+						max, _ := questions["rancher_max_version"].(string)
+						isInRange, err := isInMinMaxRange(rancherSemVer, min, max)
+						if err != nil {
+							return nil, err
+						}
+						if isInRange {
+							chartVersions = append(chartVersions, v)
+						}
+					}
+				}
+			}
```

Comment:
Would be more idiomatic to continue when co0nditions aren't met than it is to nest conditionals. Should make it easier to follow.